language-server/src/main/java/org/sonarlint/languageserver/SonarLintLanguageServer.java

```diff
@@ -570,23 +778,36 @@ public WorkspaceService getWorkspaceService() {
 
   @Override
   public CompletableFuture<Object> executeCommand(ExecuteCommandParams params) {
+    List<Object> args = params.getArguments();
     switch (params.getCommand()) {
       case SONARLINT_OPEN_RULE_DESCRIPTION_COMMAND:
-        List<Object> args = params.getArguments();
+        if (args == null) {
+          break;
+        }
         if (args.size() != 1) {
-          warn("Expecting 1 argument");
+          logger.warn("Expecting 1 argument");
         } else {
           String ruleKey = parseToString(args.get(0));
-          RuleDetails ruleDetails = engine.getRuleDetails(ruleKey);
+          // TODO use the correct engine (currently only Atom reaches this code, and it doesn't have connected mode, yet)
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.core%3Asonarlint-core-parent&pullRequest=195&issues=AWNdRMBQnxz_298731ln&open=AWNdRMBQnxz_298731ln)

[ISSUE_KEY]: AWNdRMBQnxz_298731ln